### PR TITLE
Github actions go add platforms and non-English tests

### PR DESCRIPTION
**Pull request by iliadrm**

### CODE REVIEW

File: binding/go/porcupine_test.go [hunk 1]
```diff
@@ -13,28 +13,61 @@ package porcupine
...
 	"math"
 	"os"
 	"path/filepath"
 	"testing"
+	"reflect"
 )
 
-var pvTestAccessKey string
+var (
+	pvTestAccessKey string
```
Comment (kenarsa) [inline binding/go/porcupine_test.go:26]: I know this is not part of this code change but I rather drop the `pv` here. You can use something like `testAccessKey` or even `accessKey`.

File: binding/go/porcupine_test.go [hunk 2]
```diff
@@ -13,28 +13,61 @@ package porcupine
...
+	p               Porcupine
```
Comment (kenarsa) [inline binding/go/porcupine_test.go:27]: in the previous version, `p` was local in a function and it made sense to have a short name for it. Now that it has a larger scope we prob should give it a better name. `porcupine` ...

File: binding/go/porcupine_test.go [hunk 3]
```diff
@@ -13,28 +13,61 @@ package porcupine
...
-func TestProcess(t *testing.T) {
+func appendLanguage(s string, language string) string {
+	if language == "en" {
+		return s;
+	}
+	return s + "_" + language
+}
+
+func getTestModelPath(language string) string {
+	modelPath, _ := filepath.Abs(fmt.Sprintf(
```
Comment (kenarsa) [inline binding/go/porcupine_test.go:46]: with nested calls that don't fit into a line please prefer to break them like below:

```go
relPath = fmt.Sprintf(
		"../../lib/common/%s.pv",
		appendLanguage("porcupine_params", language))
filepath.Abs(relPath)
```

File: binding/go/porcupine_test.go [hunk 4]
```diff
@@ -44,16 +77,17 @@ func TestProcess(t *testing.T) {
 	t.Logf("Frame Length: %d", FrameLength)
 	t.Logf("Sample Rate: %d", SampleRate)
 
-	data, err := ioutil.ReadFile(test_file)
+	testFile, _ := filepath.Abs(filepath.Join("../../resources/audio_samples", audioFileName))
```
Comment (kenarsa) [inline binding/go/porcupine_test.go:80]: `testAudioPath` ?

**Review by kenarsa — APPROVED** [review]

File: binding/go/porcupine_test.go [hunk 5]
```diff
@@ -44,16 +79,17 @@ func TestProcess(t *testing.T) {
 	t.Logf("Frame Length: %d", FrameLength)
 	t.Logf("Sample Rate: %d", SampleRate)
 
-	data, err := ioutil.ReadFile(test_file)
+	testAduioPath, _ := filepath.Abs(filepath.Join("../../resources/audio_samples", audioFileName))
```
Comment (laves) [inline binding/go/porcupine_test.go:82]:
```suggestion
	testAudioPath, _ := filepath.Abs(filepath.Join("../../resources/audio_samples", audioFileName))
```

**Review by laves — CHANGES_REQUESTED** [review]
I'm liking this pattern we've settled on for the new tests!

**Comment by laves** [comment]
Can you look into why Beaglebone failed?

**Review by laves — APPROVED** [review]

**Pull request merged**